experimental/casemapping/src/internals.rs

```diff
@@ -95,11 +95,12 @@ impl DotType {
     }
 }
 
-// The datatype stored in the codepoint trie for casemapping.
+/// The datatype stored in the codepoint trie for casemapping.
 #[derive(Copy, Clone, Debug, Eq, PartialEq)]
 #[cfg_attr(feature = "serde", derive(serde::Deserialize))]
-#[cfg_attr(feature = "datagen", derive(serde::Serialize))]
-pub struct CaseMappingData(u16);
+#[cfg_attr(feature = "datagen", derive(serde::Serialize, databake::Bake))]
+#[cfg_attr(feature = "datagen", databake(path = icu_casemapping::provider))]
+pub struct CaseMappingData(pub u16);
```

Comment:
Nit: keep the field private and implement `CaseMappingData::const_default()` if possible